mesonbuild/backend/backends.py

```diff
@@ -910,46 +995,49 @@ def get_mingw_extra_paths(self, target):
                 paths.update(cc.get_library_dirs(self.environment))
         return list(paths)
 
-    def determine_windows_extra_paths(self, target: T.Union[build.BuildTarget, str], extra_bdeps):
-        '''On Windows there is no such thing as an rpath.
+    def determine_windows_extra_paths(
+            self, target: T.Union[build.BuildTarget, build.CustomTarget, programs.ExternalProgram, mesonlib.File, str],
```

Comment:
I don't really know the code but from the current annotations this looks like a `build.CustomTargetIndex` or `build.GeneratedList` might also be passed here.

Reply:
I don't see any path where that could happen